### PR TITLE
searchview_added

### PR DESCRIPTION
**Pull request by AhsanBracu**

### CODE REVIEW

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 1]
```diff
@@ -139,14 +157,95 @@ private void displayView(int position) {
...
+                public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+
+                    //   Cursor cursor = (Cursor) myList.getItemAtPosition(position);
+
+                    //  String selectedName = cursor.getString(cursor.getColumnIndexOrThrow("name"));
+                    Toast.makeText(MainActivity.this, "selectedName :" + staff_id, Toast.LENGTH_LONG).show();
+                    Intent intent = new Intent(getApplication(), StaffDetail.class);
+                    intent.putExtra("ID", id);
+                    startActivity(intent);
+                     /*   myList.setAdapter(defaultAdapter);
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: remove comment out method if unuse

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 2]
```diff
@@ -139,14 +157,95 @@ private void displayView(int position) {
...
+       // mDbHelper.db_value();
+        add_data();
+        Cursor cursor = mdDatabaseHelper.searchByInputText((query != null ? query : "@@@@"));
+
+        if (cursor != null) {
+
+            String[] from = new String[]{mdDatabaseHelper.FTS_COLUMN_NAME, mdDatabaseHelper.FTS_COLUMN_PHONE,
+                    mdDatabaseHelper.FTS_BIRTH_PLACE_FIELD,mdDatabaseHelper.FTS_BIRTH_DATE_FIELD};
+
+              /*  for (int i=0;i<from.length;i++){
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: remove this please

File: app/src/main/res/layout/result_search_item.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical" android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+
```
Comment (nvmanh) [inline app/src/main/res/layout/result_search_item.xml]: remove blank line

File: app/src/main/java/com/framgia/hrm/database/DatabaseHelper.java [hunk 4]
```diff
@@ -192,13 +197,14 @@ public long createPosition(Position pos) {
      * @param searchfts
      * @return
      */
-    long insert_searchitem(SearchStaff searchfts) {
+    public long insert_searchitem(SearchStaff searchfts) {
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/database/DatabaseHelper.java]: name method convention

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 5]
```diff
@@ -139,14 +157,95 @@ private void displayView(int position) {
...
+
+                    Toast.makeText(getApplicationContext(),"From  "+from[i],Toast.LENGTH_LONG).show();
+
+                }*/
+
+
+            int[] to = new int[]{R.id.search_result_text_view, R.id.result_number, R.id.birth_place,R.id.birth_date};
+
+            SimpleCursorAdapter cursorAdapter1 = new SimpleCursorAdapter(this, R.layout.result_search_item, cursor, from, to);
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: cursorAdapter1 re-name this please

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 6]
```diff
@@ -139,14 +157,95 @@ private void displayView(int position) {
...
+            });
+
+        }
+    }
+
 
     @Override
     public boolean onQueryTextSubmit(String query) {
+        try {
+            displayResults(query + "*");
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: oh, if this query null? check it

Comment (AhsanBracu) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: there is no data yet, we are working on staff adding

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 7]
```diff
@@ -139,14 +157,95 @@ private void displayView(int position) {
...
         return false;
     }
 
     @Override
     public boolean onQueryTextChange(String newText) {
+
+        if(!newText.isEmpty()){
+
+            try {
+                displayResults(newText+"*");
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: this function block same as line 227 above, can we make 1 mothod for this?
such as:
private boolean isValidQuery(String query){
    if(TextUtils.isEmpty) return false;
    displayResults(newText+"*");
    return true;
}

Comment (AhsanBracu) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: i have to override these two methods,, are you asking me to check if the giving text is empty or not in a method?

Comment (AhsanBracu) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: private boolean isValidQuery(String query){
if(TextUtils.isEmpty) return false;
displayResults(newText+"*");
return true;
}
public boolean onQueryTextSubmit(String query) {
sValidQuery(String query);
}
public boolean onQueryTextChange(String newText) {
sValidQuery(String query);
what about this line ---(list_department.setAdapter(adapter);)
}

**Pull request merged**